lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb

```diff
@@ -267,13 +313,25 @@ def self.bump_version_release
           return verified_version
         end
 
-        # Update the `build.gradle` file with new `versionName` and `versionCode` values, both or the `defaultConfig` and `vanilla` flavors
-        #
-        # @param [Hash] new_version_beta The version hash for the beta (vanilla flavor), containing values for keys "name" and "code"
-        # @param [Hash] new_version_alpha The version hash for the alpha (defaultConfig), containing values for keys "name" and "code"
-        # @env HAS_ALPHA_VERSION If set (with any value), indicates that the project uses `vanilla` flavor.
-        #
-        def self.update_versions(new_version_beta, new_version_alpha)
+        # Update the `version.properties` file with new `versionName` and `versionCode` values
+        #
+        # @param [String] app The name of the app to be used for beta and alpha version update
+        # @param [Hash] new_version_beta The version hash for the beta, containing values for keys "name" and "code"
+        # @param [Hash] new_version_alpha The version hash for the alpha , containing values for keys "name" and "code"
+        #
+        def self.update_versions(app, new_version_beta, new_version_alpha)
+          if properties_file_exists
+            new_version_name_beta_key = "#{app}.versionName"
+            new_version_code_beta_key = "#{app}.versionCode"
+            new_version_name_alpha_key = "#{app}.alpha.versionName"
+            new_version_code_alpha_key = "#{app}.alpha.versionCode"
+            Action.sh('./gradlew', 'updateVersionProperties', "-Pkey=#{new_version_name_beta_key}", "-Pvalue=#{new_version_beta[VERSION_NAME]}")
+            Action.sh('./gradlew', 'updateVersionProperties', "-Pkey=#{new_version_code_beta_key}", "-Pvalue=#{new_version_beta[VERSION_CODE]}")
+            Action.sh('./gradlew', 'updateVersionProperties', "-Pkey=#{new_version_name_alpha_key}", "-Pvalue=#{new_version_alpha[VERSION_NAME]}") unless new_version_alpha.nil?
+            Action.sh('./gradlew', 'updateVersionProperties', "-Pkey=#{new_version_code_alpha_key}", "-Pvalue=#{new_version_alpha[VERSION_CODE]}") unless new_version_alpha.nil?
```

Comment:
We should document that this task is assumed to exist in the host app repo.

This should be documented at least:
 - in the `CHANGELOG`, to help people know how to migrate to the new system when they will migrate their host app and look at what's new
 - in each actions' documentation in which this is relevant, so that any future adopter (even one that just migrates right now, but will decide to use this new system in the future) has a reference documentation

As a bonus, it would also be nice to make the release toolkit print a message if that task doesn't exist, but I'm not sure that is gonna be easy (maybe catch the potential error of `sh` to print additional help?)

Reply:
I've both updated the Changelog and the action documentation here! Hmm I'll think a bit more about how best to fail here if the gradle task isn't present but might handle that in a smaller PR

Reply:
Yeah the failure detection might be tricky enough, totally ok to handle that in a separate PR… or not at all and consider this to be ok to fail as long as it's documented.

Reply:
Do you think it also might be useful to log another user message here? something like `attempting to update properties file with './gradlew updateVersionProperties' `

Reply:
Nah iinm `Action.sh` already logs the commands it runs in stdout so that should be enough